src/coordinator/functions/logical/common.go

```diff
@@ -66,10 +67,82 @@ func hashFunc(on bool, names ...string) func(models.Tags) uint64 {
 	return func(tags models.Tags) uint64 { return tags.IDWithExcludes(names...) }
 }
 
-// Processor is implemented by each logical transform
-type Processor interface {
-	Process(lhs block.Block, rhs block.Block) (block.Block, error)
+const initIndexSliceLength = 10
+
+var (
+	errMismatchedBounds     = errors.New("block bounds are mismatched")
+	errMismatchedStepCounts = errors.New("block step counts are mismatched")
+)
+
+func combineMetaAndSeriesMeta(
+	meta, otherMeta block.Metadata,
+	seriesMeta, otherSeriesMeta []block.SeriesMeta,
+) (block.Metadata, []block.SeriesMeta, []block.SeriesMeta, error) {
+	if !meta.Bounds.Equals(otherMeta.Bounds) {
+		return block.Metadata{},
+			[]block.SeriesMeta{},
+			[]block.SeriesMeta{},
+			errMismatchedBounds
+	}
+
+	// NB (arnikola): mutating tags in `meta` to avoid allocations
```

Comment:
I'm not sure it would be getting shared? Only one process should be using the block at a time, unless I misunderstand how we're processing downstream functions

Reply:
If a child op has two parents ?